libraries/chain/db_block.cpp

```diff
@@ -277,6 +277,8 @@ processed_transaction database::push_proposal(const proposal_object& proposal)
    size_t old_applied_ops_size = _applied_ops.size();
 
    try {
+      if( _undo_db.size() >= _undo_db.max_size() )
```

Comment:
I recommend that we just throw.

Reply:
Throwing here would prevent nested proposals, which would be a consensus change.

Reply:
Got it. Thanks.